Update internal invariants in the world of manuallyBoxedKernel

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36536 Functionalize qconcat and register with new registration API
* #36535 Convert qclamp to new operator registration API.
* #36534 Convert qbatch_norm to new operator registration API.
* #36531 Update quantized README for registering operators with new API.
* #36527 Functionalize qadd and register with new registration API.
* #36520 Add a centralized TORCH_LIBRARY declaration for quantized and xnnpack ops.
* #36513 Add explicit schema for quantized conv/conv_prepack
* #36512 Port detach/detach to new registration API
* #36505 Make it an error to TORCH_LIBRARY the same namespace twice.
* #36390 Port Resize to use new registration API.
* #36389 Update reference to RegisterOperators in error message in Convolution
* #36264 Propagate and report the file/line number of registrations on errors.
* #36258 Switch to pybind11 style registration function API.
* #36250 Unconditionally register schema even for manual registration.
* #36240 Expunge TensorId from all DispatchKey names.
* **#36388 Update internal invariants in the world of manuallyBoxedKernel**

(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

ezyang committed:
Update internal invariants in the world of manuallyBoxedKernel

(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

smessmer reviewed (changes requested):
Thanks for fixing this. We should probably do this in OperatorEntry::setManuallyBoxedKernel_ though and not in OperatorEntry::checkInvariants.

smessmer reviewed (approved):
approving to unblock fix

ezyang committed:
Update on "Update internal invariants in the world of manuallyBoxedKernel"


(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

[ghstack-poisoned]

ezyang committed:
Update on "Update internal invariants in the world of manuallyBoxedKernel"


(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

[ghstack-poisoned]

ezyang committed:
Update on "Update internal invariants in the world of manuallyBoxedKernel"


(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

[ghstack-poisoned]

ezyang committed:
Update on "Update internal invariants in the world of manuallyBoxedKernel"


(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

[ghstack-poisoned]

ezyang committed:
Update on "Update internal invariants in the world of manuallyBoxedKernel"


(This wants to make me barf).

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20964195](https://our.internmc.facebook.com/intern/diff/D20964195)

[ghstack-poisoned]
